docs/source/developing.rst

```diff
@@ -101,8 +101,36 @@ We use tools to make sure that our code is both functional and pretty. This make
 easier to work together. See instructions for the tools in `tools.rst <https://github.com/ixdat/ixdat/blob/main/TOOLS.rst>`_
 
 
+Testing
+*******
+
+Software tests are welcomed! The basic test suite is included in this
+repository in the ``tests`` directory.
+
+However, if your tests have a specific purpose and/or depend on additional
+data, then the data should be initialized in a **submodule**.  Additionally,
+the tests should be **marked** as ``external`` with the pytest mark decorator.
+It is done like this, so other developers do not have to download your data
+through submodules, when they do not wish to test their new feature with your
+tests and data.
+
+If you want to run the complete test suite that includes external tests from
+all contributing developers, you need to initialize necessary submodules::
+
+    git submodule update --init --recursive
+
+And then run command for testing with ``--external`` option::
```

Comment:
Is there a workflow whereby I have my test data in a submodule and run the tests on my new feature but don't download other testdata or run other external tests?

I think no, because there's only the `--external` tag.
Should the relevant tests not be marked with `--external` until a feature branch is merged? And when should the git submodule be created? What is the intended workflow there?

Reply:
Good question, yes there is. Executing only some external tests or a test with data from just some submodules can be achieved with pytest `-k` option -> [keyword expression](https://docs.pytest.org/en/7.1.x/how-to/usage.html#specifying-which-tests-to-run).

E.g. to run the tests from this PR only, run this: `pytest tests --external -k eclab`. In case there is more Python modules with "eclab" in their name, or when one wants to be extra specific, run `pytest tests --external -k test_eclab_parser`. Specifying an expression with `-k` works even for classes, methods and functions.